modules/olm-updating-operator-catalog-image.adoc

```diff
@@ -0,0 +1,151 @@
+...
+Uploading ... 244.9kB/s
+Pushed sha256:f73d42950021f9240389f99ddc5b0c7f1b533c054ba344654ff1edaf6bf827e3 to example_registry:5000/openshift/redhat-operators:v1
+----
+<1> Organization (namespace) to pull from an App Registry instance.
+<2> Image repository tag to apply to the built catalog image. Specify a namespace and give it a tag, for example, `v2` because it is the updated catalog.
+
+. Mirror the contents of your catalog to your target registry. The following
+`oc adm catalog mirror` command extracts the contents of your custom Operator
+catalog image to generate the manifests required for mirroring and mirrors the
+images to your registry:
++
+----
+$ oc adm catalog mirror \
+    <registry_host_name>:<port>/<namespace>/redhat-operators:v2 \ <1>
+    <registry_host_name>:<port>/openshift <2>
+
+mirroring ...
+----
+<1> Specify your new Operator catalog image.
+<2> Specify a namespace to mirror all referenced images.
++
+This command also generates the following files:
++
+--
+* The `imageContentSourcePolicy.yaml` file defines an ImageContentSourcePolicy
+object that can configure nodes to translate between the image references stored
+in Operator manifests and the mirrored registry.
+* The `mapping.txt` file contains all of the source images and where to map them
+in the target registry. This file is compatible with `oc image mirror` and can
+be used to further customize the mirroring configuration.
+--
++
+By default, these files are created in the `./manifests` directory:
++
+----
+$ ls ./manifests
+imageContentSourcePolicy.yaml
+mapping.txt
+----
+
+. Apply the manifests:
++
+----
+$ oc apply -f ./manifests
+----
++
+[IMPORTANT]
+====
+If any invalid source tags are found, remove the offending mappings from your
+`mapping.txt` file and pass the file to the `oc image mirror` command to
+continue.
+====
++
+[IMPORTANT]
+====
+It is possible that you do not need to apply the `imageContentSourcePolicy.yaml`
+manifest. Complete a `diff` of the files to determine if changes are necessary.
+====
+
+. Update your CatalogSource object that references your catalog image.
+
+.. If you have your original `catalogsource.yaml` file for this CatalogSource:
+
+... Edit your `catalogsource.yaml` file to reference your new catalog image in the
+`spec.image` field:
++
+[source,yaml]
+----
+apiVersion: operators.coreos.com/v1alpha1
+kind: CatalogSource
+metadata:
+  name: my-operator-catalog
+  namespace: openshift-marketplace
+spec:
+  sourceType: grpc
+  image: <registry_host_name>:<port>/<namespace>/redhat-operators:v2 <1>
+  displayName: My Operator Catalog
+  publisher: grpc
+----
+<1> Specify your new Operator catalog image.
+
+... Use the updated file to replace the CatalogSource object:
++
+----
+$ oc replace -f catalogsource.yaml
+----
+
+.. Alternatively, edit the CatalogSource using the following command and reference
```

Comment:
Added this in case they didn't have the file hanging around still?

Reply:
Could we use a patch instead?

Reply:
Oh, yes that would be better!

Reply:
I'll do this in a follow-up in the interest of getting this merged and published today since we've got the various acks already.